packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -387,6 +407,44 @@ class Auth {
         })
     }
 
+    async refreshAccessToken() {
+        const refreshTokenRegistered = this.get('refresh_token_registered')
+        const refreshTokenGuest = this.get('refresh_token_guest')
+        const refreshToken = refreshTokenRegistered || refreshTokenGuest
+        if (refreshToken) {
+            try {
+                const result = await this.queueRequest(
+                    () =>
+                        helpers.refreshAccessToken(
+                            this.client,
+                            {
+                                refreshToken,
+                                ...(this.defaultDnt !== undefined && {dnt: this.defaultDnt})
+                            },
+                            {
+                                clientSecret: this.clientSecret
+                            }
+                        ),
+                    !!refreshTokenGuest
+                )
+                this.handleTokenResponse(result, !!refreshTokenGuest)
+                return result
+            } catch (error) {
+                // If the refresh token is invalid, we need to re-login the user
+                if (error instanceof Error && 'response' in error) {
+                    // commerce-sdk-isomorphic throws a `ResponseError`, but doesn't export the class.
+                    // We can't use `instanceof`, so instead we just check for the `response` property
+                    // and assume it is a fetch Response.
```

Comment:
This is a side effect of "overly fat try/catch blocks". If you don't know which of two functions errored and you need to handle for both, you should have a separate try/catch.

Let's do that here. `try/catch` once wrapping the `const result = await this.queueResuest()` line. 

And then a second `try/catch` for `this.handleTokenResponse()`